src/utils/migration.ts

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { encrypt } from '@alephium/sdk/dist/lib/password-crypto'
+
+export const latestUserDataVersion = '2022-05-27T12:00:00Z'
+
+//
+// Arguments are essentially dependencies of the migrations.
+// Unfortunately mnemonics are a dependency due to being needed to encrypt address
+// metadata, so migration must absolutely occur after login at the earliest.
+//
+// Future dependencies must be explained as they are added.
+//
+export const migrateUserData = (mnemonic: string) => {
```

Comment:
Do we want to put this in `wallet-sdk`?

Also better to have unit tests cover this function. It can be altered accidentally.

Reply:
Yeah having tests would've caught the error @nop33 hit right away too. I'll do this next week along with the other tests :ok_hand:

Reply:
> Do we want to put this in wallet-sdk?

We can move it there at a later time